### PR TITLE
Feat/cart

### PR DESCRIPTION
**Pull request by manish591**
#### Features Added
* Add Item To Cart from product listing page, after clicking button text will change to `go to cart`. 
* Fetched cart data from server using Axios and useeffect 
* Remove Item from cart
* Checkout flow in sync with cart data
* Update the quantity of cart items, either increase the quantity or decrease.

https://user-images.githubusercontent.com/79859472/159172616-a5ca9489-5015-43d8-ab3b-b4cb0dc16398.mp4

### CODE REVIEW

File: src/pages/cart/Cart.css [hunk 1]
```diff
@@ -276,6 +276,29 @@
...
+  cursor: pointer;
+}
+
+.card__update-quantity--add:disabled, 
+.card__update-quantity--remove:disabled {
+  cursor: not-allowed;
+}
+
+.cart-icon-quantity {
+  font-size: .8rem !important;
```
Comment (rohit8686) [inline src/pages/cart/Cart.css:299]: Better to avoid using !important, other than these everything looks good

**Review by rohit8686 — APPROVED** [review]

File: src/pages/wishlist/Wishlist.jsx [hunk 2]
```diff
@@ -7,27 +7,27 @@ import { useAuth } from "../../hooks";
 
 const Wishlist = () => {
   const { state, stateDispatch } = useStateContext();
+  const { wishlistData } = state;
   const { currentUser } = useAuth();
   const [isLoading, setIsLoading] = useState(true);
+  const [myWishlist, setMyWishlist] = useState([]);
 
   useEffect(() => {
+    console.log('useeffect runs');
```
Comment (Shah-Arsalan) [inline src/pages/wishlist/Wishlist.jsx:16]: Remove console logs from your code

**Review by Shah-Arsalan — APPROVED** [review]

File: src/pages/wishlist/Wishlist.jsx (hunk 2 above)
Comment (manish591) [reply src/pages/wishlist/Wishlist.jsx:16]: Thanks for pointing it out.

File: src/pages/cart/Cart.css (hunk 1 above)
Comment (manish591) [reply src/pages/cart/Cart.css:299]: Using this I'm overriding material icon default styling. It is not applying without `important`.

Comment (rohit8686) [reply src/pages/cart/Cart.css:299]: try increasing specificity by adding another class for icon styling. i believe it will work

**Pull request merged**